source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -162,6 +209,13 @@ void TextureBaker::optimizeBakedTextures(NodePtr shader)
         return;
     }
 
+    // Early out if not optimizing		
```

Comment:
Wouldn't this early out prevent the detection of uniform images when _optimizeConstants is set to true?  Ideally we still want to optimize individual UDIMs within a UDIM set, even when baking content for applications that don't support constant values.